src/test/deploy/functions/prepare.spec.ts

```diff
@@ -159,4 +167,71 @@ describe("prepare", () => {
       expect(beforeSignIn.blockingTrigger.options?.refreshToken).to.be.false;
     });
   });
+
+  describe.skip("prepare", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+
+  describe.skip("preparePhase1", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+  describe.skip("preparePhase2", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+  describe.skip("preparePhase3", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+  describe.skip("preparePhase4", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+  describe.skip("preparePhase5", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+
+  describe.skip("preparePhase6", () => {
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    it.skip("should be tested", () => {});
+  });
+
+  describe("preparePhase7", () => {
+    it("does not call applyBackendHashToBackends when preview flag is disabled", async () => {
+      const applyBackendHashToBackends = sinon.stub(applyHash, "applyBackendHashToBackends");
+
+      const context: Context = {
+        projectId: "projectId",
+      };
+      const wantBackends: Record<string, backend.Backend> = {
+        backend1: backend.empty(),
+        backend2: backend.empty(),
+      };
+      previews.skipdeployingnoopfunctions = false;
```

Comment:
You may want to save and restore `previews.skipdeployingnoopfunctions` in a `before/after` in the test file. Accidentally leaving this in an undefined state (which _might_ be `true` if the below test is run next) can result in other tests failing in unexpected ways.

Reply:
Good point! Updated!